### PR TITLE
Text input component

### PR DESCRIPTION
**Pull request by ahmedkhitaby**
- [x] created TextInput component with it's variant and inputType props and initializing their types
- [x] tested component with data attributes instead of  snapshot testing

### CODE REVIEW

File: modules/shared/components/TextInputs/TextInput.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+          {variants === ETextInput.Variants.Error && <ErrorIcon />}
+          {variants === ETextInput.Variants.Success && <SuccessIcon />}
+        </span>
+        <span className={styles['control-icon']}>
+          {inputType === ETextInput.InputType.RightIcon && <ArrowDownIcon />}
+          {inputType === ETextInput.InputType.LeftIcon && <HomeIcon />}
+          {inputType === ETextInput.InputType.Prefix && '+20'}
+          {inputType === ETextInput.InputType.PrefixDropdown && (
+            <select
+              className="bg-transparent focus:outline-none"
```
Comment (AhmedAdel999) [inline modules/shared/components/TextInputs/TextInput.tsx:86]: Remove the classNames and add them using the css module file.

File: modules/shared/components/TextInputs/types/ITextInputs.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import type * as ETextInput from './ETextInput';
+
+declare namespace ITextInputs {
+  export interface IProps {
+    [property: string]: unknown;
+    label: string;
```
Comment (AhmedAdel999) [inline modules/shared/components/TextInputs/types/ITextInputs.d.ts:6]: Label should be optional.

**Review by AhmedAdel999 — CHANGES_REQUESTED** [review]

**Comment by atwfk** [comment]
added the needed adjustments

**Comment by Ali-Khamis** [comment]
i have some notices :

1. when the input is disabled the icon still shown and not even blured ( the clear icon function even when the input disabled ) and the border color keeps it's color Screenshot from 2021-05-16 18-30-52 
2. the space doesn't match the design Screenshot from 2021-05-16 18-34-13
3. when writing the text doesn't stop before the icon but it goes under it
Screenshot from 2021-05-16 18-41-16

**Comment by ahmedkhitaby** [comment]
@Ali-Khamis the required changes have been updated

File: modules/shared/components/atoms/TextInputs/TextInput.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  );
+
+  /*
+    when it comes to authentication we need to validate
+  */
+  // const inputId = id.trim().split(' ').join('');
+
+  return (
+    <div className={styles['form-group']}>
+      {label !== undefined && (
```
Comment (Ali-Khamis) [inline modules/shared/components/atoms/TextInputs/TextInput.tsx:58]: this condition doesn't work like it should 
if there is no label giving it stills have a label element rendered
Screenshot from 2021-05-16 23-28-13

**Comment by Ali-Khamis** [comment]
on right icon when disabled the  arrow down color doesn't match the design 
Screenshot from 2021-05-16 23-32-41

**Comment by Ali-Khamis** [comment]
1-on left icon when disabled and hovered the icon color changes
Screenshot from 2021-05-16 23-36-05

2-consider adding ``cursor-not-allowed`` when the input is disabled

**Comment by Ali-Khamis** [comment]
LGTM

**Comment by AhmedAdel999** [comment]
LGTM

**Review by AhmedAdel999 — APPROVED** [review]

**Pull request merged**